File: cp-trials/src/test/java/in/shabhushan/cp_trials/math/fundamental/FibonacciTest.java
package in.shabhushan.cp_trials.math.fundamental;

import org.junit.Test;

import static in.shabhushan.cp_trials.math.fundamental.Fibonacci.fib;
import static in.shabhushan.cp_trials.math.fundamental.Fibonacci.fibTailRec;
import static org.junit.Assert.assertEquals;

public class FibonacciTest {
  @Test
  public void fibonacci() {
    int fib = fib(32);

    assertEquals(3524578, fib);
    assertEquals(7049155, Fibonacci.counter);
  }

  @Test
  public void testFibonacciTailRec() {
    int fib = fibTailRec(32);

    assertEquals(3524578, fib);
    assertEquals(33, Fibonacci.tailRecCounter);
  }
}

File: cp-trials/src/test/kotlin/in/shabhushan/cp_trials/dsbook/chapter2-recursion/FibonacciTest.kt
package `in`.shabhushan.cp_trials.dsbook.`chapter2-recursion`

import org.junit.Assert.assertArrayEquals
import kotlin.test.Test

class FibonacciTest {
  @Test
  fun testFibonacci() {
    val result = fibonacciSequence(0, 1, 10)

    assertArrayEquals(arrayOf(0, 1, 1, 2, 3, 5, 8, 13, 21, 34), result.toTypedArray())
  }
}
